aw/ViewControllers/Reaches/FavoriteListTableViewController.swift

```diff
@@ -1,22 +1,40 @@
 import UIKit
 
 class FavoriteListTableViewController: RunListTableViewController {
+    let UPDATE_INTERVAL_s: Double = -3600
+
     override func viewDidLoad() {
         predicates.append(NSPredicate(format: "favorite = TRUE" ))
         super.viewDidLoad()
         favorite = true
     }
 
-    override func refreshReaches(sender: UIRefreshControl) {
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        print("appeared")
+        guard let favoritesLastUpdated = DefaultsManager.favoritesLastUpdated
+            else { return }
+
+        if favoritesLastUpdated < Date().addingTimeInterval(UPDATE_INTERVAL_s) {
+            refreshReaches(sender: nil)
+        }
+    }
+
+    override func refreshReaches(sender: UIRefreshControl?) {
         guard let fetchedResultsController = fetchedResultsController,
-            let reaches = fetchedResultsController.fetchedObjects
+            let reaches = fetchedResultsController.fetchedObjects,
+            let refreshControl = self.tableView.refreshControl
             else { return }
 
         let reachesIds = reaches.map { String($0.id) }
 
+        refreshControl.beginRefreshing()
+        self.tableView.setContentOffset(CGPoint(x: 0, y: -refreshControl.frame.height), animated: true)
```

Comment:
Why do you need to set this? Seems like this should go in initialization since it's not unset when refreshing finishes?

Reply:
The refresh controller hides itself when it gets .endRefreshing, but it doesn't show itself (as it expects that you've dragged it down and exposed it). So if we want to show the refreshController when we trigger a refresh another way, we need to show it.

Reply:
Hmm, pretty sure there's different way to do this, but I'll leave that up to you.